lib/Service/DashboardService.php

```diff
@@ -0,0 +1,494 @@
+<?php
+/**
+ * OpenRegister Dashboard Service
+ *
+ * This file contains the service class for handling dashboard related operations
+ * in the OpenRegister application.
+ *
+ * @category Service
+ * @package  OCA\OpenRegister\Service
+ *
+ * @author    Conduction Development Team <info@conduction.nl>
+ * @copyright 2024 Conduction B.V.
+ * @license   EUPL-1.2 https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * @version GIT: <git_id>
+ *
+ * @link https://www.OpenRegister.app
+ */
+
+namespace OCA\OpenRegister\Service;
+
+use OCA\OpenRegister\Db\RegisterMapper;
+use OCA\OpenRegister\Db\SchemaMapper;
+use OCA\OpenRegister\Db\ObjectEntityMapper;
+use OCA\OpenRegister\Db\AuditTrailMapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class DashboardService
+ *
+ * Service for handling dashboard related operations.
+ *
+ * @package OCA\OpenRegister\Service
+ */
+class DashboardService
+{
+    /**
+     * Register mapper instance for handling register operations.
+     *
+     * @var RegisterMapper The register mapper instance.
+     */
+    private readonly RegisterMapper $registerMapper;
+
+    /**
+     * Schema mapper instance for handling schema operations.
+     *
+     * @var SchemaMapper The schema mapper instance.
+     */
+    private readonly SchemaMapper $schemaMapper;
+
+    /**
+     * Object entity mapper instance for handling object operations.
+     *
+     * @var ObjectEntityMapper The object entity mapper instance.
+     */
+    private readonly ObjectEntityMapper $objectMapper;
+
+    /**
+     * Audit trail mapper instance for handling audit trail operations.
+     *
+     * @var AuditTrailMapper The audit trail mapper instance.
+     */
+    private readonly AuditTrailMapper $auditTrailMapper;
+
+    /**
+     * Database connection instance.
+     *
+     * @var IDBConnection The database connection instance.
+     */
+    private readonly IDBConnection $db;
+
+    /**
+     * Logger instance for logging operations.
+     *
+     * @var LoggerInterface The logger instance.
+     */
+    private readonly LoggerInterface $logger;
+
+    /**
+     * Constructor for DashboardService
+     *
+     * @param RegisterMapper    $registerMapper   The register mapper instance
+     * @param SchemaMapper     $schemaMapper    The schema mapper instance
+     * @param ObjectEntityMapper $objectMapper    The object entity mapper instance
+     * @param AuditTrailMapper  $auditTrailMapper The audit trail mapper instance
+     * @param IDBConnection     $db              The database connection instance
+     * @param LoggerInterface   $logger          The logger instance
+     *
+     * @return void
+     */
+    public function __construct(
+        RegisterMapper $registerMapper,
+        SchemaMapper $schemaMapper,
+        ObjectEntityMapper $objectMapper,
+        AuditTrailMapper $auditTrailMapper,
+        IDBConnection $db,
+        LoggerInterface $logger
```

Comment:
Please call these with `private readonly RegisterMapper $registerMapper`, etc, that saves actually quite a lot of LOC

Reply:
Done